interface/main/messages/js/reminder_appts.js

```diff
@@ -325,22 +348,139 @@ function show_this(colorish='') {
     var pidRE = new RegExp(pidV, 'i');
     var pnameV = $("#form_patient_name").val();
     var pnameRE = new RegExp(pnameV, 'i');
+    var fromDate = $("#form_from_date").val();
+    var toDate = $("#form_to_date").val();
+    var fromDateObj = fromDate ? parseDate(fromDate) : null;
+    var toDateObj = toDate ? parseDate(toDate) : null;
+
+    $('.ALL').hide();
 
-    $('.ALL').hide().filter(function () {
+    var visibleRows = $('.ALL').filter(function () {
         var d = $(this).data();
         meets_fac = (facV === '') || (facV == d.facility);
         meets_prov = (provV === '') || (provV == d.provider);
         meets_pid = pidV === '';
+
         if ((pidV > '') && pidRE.test(d.pid)) {
             meets_pid = true;
         }
+
         meets_pname = pnameV === '';
+
         if ((pnameV > '') && pnameRE.test(d.pname)) {
             meets_pname = true;
         }
-        meets_color = (colorish === '') || (colorish == d.status );
-        return meets_fac && meets_prov && meets_pid && meets_pname && meets_color;
-    }).show('4000', 'linear');
+
+        meets_color = (colorish === '') || (colorish == d.status);
+        meets_date = true;
+
+        if (fromDateObj || toDateObj) {
+            var apptDateText = $(this).find('.appt_date').text().trim();
+            var apptDate = extractDateFromText(apptDateText);
+            
+            if (apptDate) {
+                if (fromDateObj && apptDate < fromDateObj) {
+                    meets_date = false;
+                }
+                
+                if (toDateObj && apptDate > toDateObj) {
+                    meets_date = false;
+                }
+            }
+        }
+        
+        return meets_fac && meets_prov && meets_pid && meets_pname && meets_color && meets_date;
+    });
+
+    visibleRows.show('400', 'linear');
+
+    if (visibleRows.length === 0) {
+        if ($(".alert.alert-info:contains('No Recalls Found')").length > 0) {
+            $(".alert.alert-info:contains('No Recalls Found')").show();
+        } else {
+            $("#show_recalls").prepend(
+                '<div class="alert alert-info text-center">No Recalls Found</div>'
+            );
+        }
+
+        $(".table-responsive").hide();
+    } else {
+        $(".alert.alert-info:contains('No Recalls Found')").hide();
```

Comment:
Again, better to add an id or a class name versus relying on the text as language translation will change this.